apps/backend/src/utils/errors.ts

```diff
@@ -0,0 +1,41 @@
+import { ResultAsync } from 'neverthrow';
+
+// We could define aliases for each module we got
+// There are two ways of doing this:
+// 1. Create a specific Error type
+
+// type DbError = {
+//     code: string;
+//     message: string;
+// }
+
+// ResultAsync<DB, DbError>;
+// ..., errorContext: E (DbError)
+
+//Or, create a type alias that wraps ResultAsync.
+// type ProductError =
+//   | {
+//   type: "CreateProductError",
+//   message: "Error creating product"
+// } | {
+//   type: "FindProductError",
+//   message: "Error finding a product"
```

Comment:
add this in our libs/shared directory please. so that it would be shared between all of our apps.